src/chainstate/stacks/index/storage.rs

```diff
@@ -965,6 +1646,69 @@ impl<'a, T: MarfTrieId> TrieStorageTransaction<'a, T> {
         Ok(ret)
     }
 
+    /// Run `cls` with a mutable reference to the inner trie blobs opt.
+    fn with_trie_blobs<F, R>(&mut self, cls: F) -> R
+    where
+        F: FnOnce(&Connection, &mut Option<&mut TrieFile>) -> R,
+    {
+        let mut blobs = self.blobs.take();
+        let res = cls(&self.db, &mut blobs);
+        self.blobs = blobs;
+        res
+    }
+
+    /// Store a serialized trie, given the flush options and its block hash.
+    /// Depending on how the trie storage is configured, the trie will either get dumped into a
+    /// blob in the underlying sqlite database, or it will be appended to a flat file that lives
+    /// alongside the sqlite database.
+    fn inner_store_trie_blob(
```

Comment:
It seems like this method is only ever invoked by `inner_flush`, and the structure of the method uses the same match on `FlushOptions`, would it make more sense if this was just folded into the `inner_flush` method?

Reply:
Sure, I can do that.